boards/pjrc/teensy4/doc/index.rst

```diff
@@ -204,23 +229,113 @@ Programming and Debugging
+
+.. tabs::
+
+   .. group-tab:: Teensy 4.0
+
+      .. zephyr-app-commands::
+         :zephyr-app: samples/basic/blinky
+         :board: teensy40
+         :goals: build
+         :compact:
+
+   .. group-tab:: Teensy 4.1
+
+      .. zephyr-app-commands::
+         :zephyr-app: samples/basic/blinky
+         :board: teensy41
+         :goals: build
+         :compact:
 
-.. _Teensy flash tools:
-   https://www.pjrc.com/teensy/loader.html
+#. Connect the board to your host computer using USB.
 
-Debugging
-=========
-Console output is mapped to teensy pins 0 (RX1) and 1 (TX1). Connect a usb-to-serial adapter
-to use this serial console. Use the following settings with your serial terminal of choice (minicom, putty,
-etc.):
+#. Tap the reset button to enter bootloader mode.
+   Red LED blinks.
 
-- Speed: 115200
-- Data: 8 bits
-- Parity: None
-- Stop bits: 1
+#. Flash the image.
+
+.. tabs::
+
+   .. group-tab:: Teensy 4.0
+
+      .. zephyr-app-commands::
+         :zephyr-app: samples/basic/blinky
+         :board: teensy40
+         :goals: flash
+         :compact:
+
+   .. group-tab:: Teensy 4.1
+
+      .. zephyr-app-commands::
+         :zephyr-app: samples/basic/blinky
+         :board: teensy41
+         :goals: flash
+         :compact:
+
+#. You should see the orange LED blink.
+
+Configuring a Console
+=====================
+
+.. tabs::
+
+   .. group-tab:: UART-Console
+
+      By default console output is mapped to teensy pins 0 (RX1) and 1 (TX1). Connect a usb-to-serial adapter
+      to use this serial console. Use the following settings with your serial terminal of choice (minicom, putty,
+      etc.):
+
+      - Speed: 115200
+      - Data: 8 bits
+      - Parity: None
+      - Stop bits: 1
+
+   .. group-tab:: USB-Console
+
+      By mapping the console output to USB, a usb-to-serial adapter is no longer required.
+      At minimum this needs the following steps. See :ref:`usb_device_stack` for more detailed information.
+
+      #. Add these lines to the ``prj.conf`` file.
+
+         .. code-block:: kconfig
+
+            CONFIG_USB_DEVICE_STACK=y
+            CONFIG_USB_CDC_ACM=y
+
+      #. Add a overlay file to project, e.g. ``teensy41.overlay``.
+
+         .. code-block:: devicetree
+
+            / {
+               chosen {
+                  zephyr,console = &cdc_acm_uart0;
+               };
+            };
+
+            &zephyr_udc0 {
+               cdc_acm_uart0: cdc_acm_uart0 {
+                  compatible = "zephyr,cdc-acm-uart";
+               };
+            };
+
+      #. Extend application ``main.c``.
+
+         .. code-block::
+
+            #include <zephyr/usb/usb_device.h>
+            ...
+            usb_enable(NULL);
+
+      #. After application startup a serial device named like
+         ``tty.usbmodem14201 - ZEPHYR`` should appear on your host computer.
+         You can use e.g. ``Serial Monitor`` plugin for VScode to monitor.
```

Comment:
sounds like a job for https://docs.zephyrproject.org/latest/snippets/cdc-acm-console/README.html?
It looks like you could drop most/all of this and just mention that people can use that snippet if they care about USB console?

Reply:
this snippet would be absolutely great, but building
`west build -p -b teensy40 -S cdc-acm-console samples/basic/blinky`

fails with a missing include anywhere in the NXP-Hal:
```
phy.c.obj.d -o modules/hal_nxp/hal_nxp/CMakeFiles/..__modules__hal__nxp.dir/mcux/middleware/mcux-sdk-middleware-usb/phy/usb_phy.c.obj -c /Users/bernhardkramer/zephyrlatest/modules/hal/nxp/mcux/middleware/mcux-sdk-middleware-usb/phy/usb_phy.c
In file included from /Users/bernhardkramer/zephyrlatest/modules/hal/nxp/mcux/middleware/mcux-sdk-middleware-usb/phy/usb_phy.c:9:
/Users/bernhardkramer/zephyrlatest/modules/hal/nxp/mcux/middleware/mcux-sdk-middleware-usb/include/usb.h:14:10: fatal error: fsl_os_abstraction.h: No such file or directory
   14 | #include "fsl_os_abstraction.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

still searching for the reason, any idea?

Reply:
updating my repo solved the problem, but still no luck with the snippet

Reply:
Reworked this section to make use of this snippet. Especially no application code changes are necessary to enable cdc-acm-console.

Behavior is also explained here:
https://github.com/zephyrproject-rtos/zephyr/discussions/53182#discussioncomment-4447048

Reply:
Please don't resolve conversations yourself, it's up to the person requesting the change to come back to it and resolve themselves. Having contributors resolving the conversation themselves just makes this task harder :)

Reply:
(see https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#workflow-suggestions-that-help-reviewers)